File: app/src/main/java/com/roy/a/FragmentPagerAdapter.java
package com.roy.a;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.roy.R;
import com.roy.ui.FApps;
import com.roy.ui.FLens;
import com.roy.ui.FSettings;

public class FragmentPagerAdapter extends FragmentStatePagerAdapter {

    private static final int NUM_PAGES = 3;

    private final Context mContext;

    public FragmentPagerAdapter(FragmentManager fragmentManager, Context context) {
        super(fragmentManager);
        mContext = context;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return FLens.newInstance();
            case 1:
                return FApps.newInstance();
            case 2:
                return FSettings.newInstance();
        }
        return new Fragment();
    }

    @Override
    public int getCount() {
        return NUM_PAGES;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return mContext.getResources().getString(R.string.tab_lens);
            case 1:
                return mContext.getResources().getString(R.string.tab_apps);
            case 2:
                return mContext.getResources().getString(R.string.tab_settings);
        }
        return super.getPageTitle(position);
    }
}

File: app/src/main/java/com/roy/a/FragmentPagerAdapter.kt
package com.roy.a

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.roy.R
import com.roy.ui.FApps
import com.roy.ui.FLens
import com.roy.ui.FSettings

class FragmentPagerAdapter(
    fragmentManager: FragmentManager,
    private val mContext: Context
) : FragmentStatePagerAdapter(fragmentManager) {

    companion object {
        private const val NUM_PAGES = 3
    }

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return FLens.newInstance()
            1 -> return FApps.newInstance()
            2 -> return FSettings.newInstance()
        }
        return Fragment()
    }

    override fun getCount(): Int {
        return NUM_PAGES
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return mContext.resources.getString(R.string.tab_lens)
            1 -> return mContext.resources.getString(R.string.tab_apps)
            2 -> return mContext.resources.getString(R.string.tab_settings)
        }
        return super.getPageTitle(position)
    }
}
